File: 0020/main.go
package main

func isValid(s string) bool {
	length := len(s)
	if length <= 0 || length%2 != 0 {
		return false
	}

	hashmap := map[byte]byte{
		')': '(',
		']': '[',
		'}': '{',
	}

	stack := []byte{}
	var lastIndex, i int

	for i = 0; i < length; i++ {
		lastIndex = len(stack) - 1
		if lastIndex <= -1 || stack[lastIndex] != hashmap[s[i]] {
			stack = append(stack, s[i])
			continue
		}
		stack = stack[:lastIndex]
	}
	return len(stack) == 0
}

File: 0020/main.swift
class Solution {
    func isValid(_ s: String) -> Bool {
        let length = s.count
        if length == 0 || length % 2 != 0{
            return false
        }

        let hashmap: [Character: Character] = [
            ")": "(",
            "]": "[",
            "}": "{",
        ]

        var stack = [Character]()
        var lastIndex = 0

        for char in s{
            lastIndex = stack.count - 1
            if lastIndex <= -1 || stack[lastIndex] != (hashmap[char] ?? "_"){
                stack.append(char)
                continue
            }
            stack.removeLast()
        }
        return stack.count == 0
    }
}